File: java/src/main/java/com/soumc/models/AddressBookCreator.java
package com.soumc.models;

public class AddressBookCreator {

	public AddressBookProtos.Person person(int id) {
		AddressBookProtos.Person.Builder person = AddressBookProtos.Person.newBuilder();
		person.setId(id);
		person.setName("full name");
		person.setEmail("a@a.com");
		AddressBookProtos.Person.PhoneNumber.Builder phone = AddressBookProtos.Person.PhoneNumber
				.newBuilder().setNumber("0000000000");
		phone.setType(AddressBookProtos.Person.PhoneType.MOBILE);
		person.addPhones(phone.build());
		return person.build();
	}
}

File: protobuf-ruby/lib/models/address_book_creator.rb
# frozen_string_literal: true

# Address book creator
class AddressBookCreator
  class << self
    def address_book(number_of_person)
      Proto::AddressBook.new(
        people: (1..number_of_person).map { |i| person(i) }
      )
    end

    def address_book_from_hash(address_h)

      people = address_h[:people].map do |person|
        phones_array = person[:phones].map do |phone|
          Proto::Person::PhoneNumber.new(
            number: phone[:number],
            type: phone[:type]
          )
        end

        Proto::Person.new(
          id: person[:id],
          name: person[:name],
          email: person[:email],
          phones: phones_array
        )
      end

      Proto::AddressBook.new(
        people: people
      )
    end

    private

    def person(id)
      Proto::Person.new(
        id: id,
        name: 'full name',
        email: 'abc@abc.com',
        phones: phones
      )
    end

    def phones
      [
        Proto::Person::PhoneNumber.new(
          number: '0000000000',
          type: Proto::Person::PhoneType::MOBILE
        )
      ]
    end
  end
end
